### DISCUSSION TITLE
Why the ansible-freeipa user module does not allow to obtain info about a user without affecting th current user state?

### DISCUSSION THREAD

**Discussion by dafizscq** [freeipa/ansible-freeipa#1230] [Ideas]
The thing is that if you want to modify information about a user, with the ansible module, only in the case that this user is enabled, I think is not possible to let the user in the same state that it was before.

I mean, if I want to change any user parameter only if the user is in state "enabled", the ipauser module will affect the current state. If an admin have changed it before to "disabled" (for instance), but I ignore that, the module will change the state to "present" (user enabled) unless you know previously that the state was "disabled", and you put that explicitly in the playbook.

I think it could be helpful to have an option in the module to retrieve all information for a user, without changing the current state, even if you don't know the state before using it.
Or if there where an state for "preserve state" (don't do anything with the current state).

It's maybe a dumb question... or does it was a developement  decision for any security reason?
Thanks in advance.

**Comment by t-woerner** [comment]
There was work started some time ago to add a query mode with state:query. But this was not completed yet.
https://github.com/freeipa/ansible-freeipa/pull/782

**Comment by t-woerner** [comment]
The issue is that the returned information could be huge as it is possible to have mass multi user tasks.
